libsolidity/interface/CompilerStack.cpp

```diff
@@ -986,7 +986,8 @@ Json::Value CompilerStack::assemblyJSON(std::string const& _contractName) const
 	Contract const& currentContract = contract(_contractName);
 	if (currentContract.evmAssembly)
 		return currentContract.evmAssembly->assemblyJSON(sourceIndices());
-	return Json::Value();
+	else
```

Comment:
Why that? Isn't the `else` unnecessary?

Reply:
And as to the style itself, IMO the pattern does not universally make the code better. It's best used when you have a clear main case and a special case. You deal with the special case first and then continue with the default one without extra nesting. Here it's the opposite, which is not necessarily an improvement over the `else`.